server/src/main/java/org/opensearch/cluster/ClusterManagerNodeChangePredicate.java

```diff
@@ -41,9 +41,13 @@
  *
  * @opensearch.internal
  */
-public final class ClusterManagerNodeChangePredicate {
+// TODO: add final keyword to the class and private keyword to the default constructor,
+// after removing the deprecated class MasterNodeChangePredicate.
+// Removed the final keyword temporarily only for making the class MasterNodeChangePredicate as a subclass,
+// so that preserving the both class names by maintaining one class implementation for backwards compatibility.
```

Comment:
I believe this class only exists as a container for a single static method. Assuming that's right, the less intrusive approach may be just to do the following in the other class (as opposed to using inheritance):

```java
@Deprecated
public final class MasterNodeChangePredicate {
  private MasterNodeChangePredicate() { }

  public static Predicate<ClusterState> build(ClusterState currentState) {
    return ClusterManagerNodeChangePredicate.build(clusterState);
  }
}
```

Because it means you don't have to change this class at all.

Reply:
Thank you so much for providing the suggestion to me! Let me check this approach. 😄